### ISSUE TITLE
hardTimeToSoftTime

### ISSUE THREAD

**Issue by mingyux** [Markbess/boteye_ros#1]
https://github.com/Markbess/boteye_ros/blob/master/src/ros_node.cpp#L90

First of all, awesome work for adding ROS support in boteye_driver.
We notice an issue that the hardware clock actually drifts from the real world clock, which means that hardTimeToSoftTime will eventually drift away.  An simple hack will be periodically *update* the soft_time_begin and hard_time_begin.

**Comment by TAO270** [comment]
@mingyux 
Yes, I also  noticed this problem. But keeping  soft_time_begin and hard_time_begin unchanged could make the time difference between continuous images constant. When updating the two initial values, it may cause other issues, such as negative time interval between continuous images.
